src/pages/ProductsPage.js

```diff
@@ -1,8 +1,16 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import styled from 'styled-components';
 import { Filters, ProductList, Sort, PageHero } from '../components';
+import { useProductsContext } from '../context/products_context';
 
 const ProductsPage = () => {
+  const { ensureProductsLoaded } = useProductsContext();
+  
+  // When the ProductsPage component is first loaded, double check that we've fetched products.
+  useEffect(() => {
+    ensureProductsLoaded()
+  }, [])
```

Comment:
I noticed that if you just go directly to the products page, rather than clicking the nav from the "Home" page (ie: if you type into your url http://localhost:3000/products), it would fail to load because the products had not been fetched yet. So here we call a function that will check if products have been loaded already, if not it will fetch them.